uni/lib/model/utils/time/weekday_mapper.dart

```diff
@@ -0,0 +1,153 @@
+/// A class that maps weekdays from one system to another.
+///
+/// ---
+/// # Explanation
+///
+/// Consider the following systems:
+///
+/// | System | Mon | Tue | Wed | Thu | Fri | Sat | Sun |
+/// |--------|-----|-----|-----|-----|-----|-----|-----|
+/// | A      | 1   | 2   | 3   | 4   | 5   | 6   | 7   |
+/// | B      | 0   | 1   | 2   | 3   | 4   | 5   | 6   |
+/// | C      | 1   | 2   | 3   | 4   | 5   | 6   | 0   |
+/// | D      | 5   | 6   | 0   | 1   | 2   | 3   | 4   |
+/// | E      | 1   | 2   | 3   | 4   | 5   | -1  | 0   |
+///
+/// All of these systems are valid and used in different contexts. This class
+/// allows for mapping weekdays from one system to another.
+///
+/// As you can see, a system is defined by two parameters: the number of the
+/// first day of the week and the number of Monday (can be another day as long
+/// as it is the same across all systems).
+class WeekdayMapper {
+  const WeekdayMapper({
+    required int fromStart,
+    required int fromMonday,
+    required int toStart,
+    required int toMonday,
```

Comment:
Can we suffix all these names with `Index`? Would save me some seconds of documentation reading.

Reply:
They are not indices. They can be -1, 7, 0, any value really

Reply:
I see.